pkg/volume/util/metrics.go

```diff
@@ -62,6 +71,18 @@ func registerMetrics() {
 	prometheus.MustRegister(storageOperationMetric)
 	prometheus.MustRegister(storageOperationErrorMetric)
 	prometheus.MustRegister(storageOperationStatusMetric)
+	prometheus.MustRegister(storageOperationEndToEndLatencyMetric)
+}
+
+func RecordOperationMetric(plugin, operationName string, secondsTaken float64, err error) {
+	status := statusSuccess
+	if err != nil {
+		storageOperationErrorMetric.WithLabelValues(plugin, operationName).Inc()
```

Comment:
We should not update the error and status metrics that are already updated as part of OperationCompleteHook. This will cause those metric to be double counted.

Reply:
It will actually not for error count but will double count for status.
the entry point of this call will be from persistentvolume/metrics/metrics.go (see another metrics.go file in this PR)
whenever there's an error, it will NOT call this function to report. I put the status/error count metric here for reasons bellow, LMK what do you think:
For external provisioner/deleter workflow, OperationCompleteHook wont be able to capture errors/status count, it makes sense to remove OperationCompleteHook from pv_controller in another PR to fully utilize RecordOperationMetric in the future. I am a bit hesitate to make the change in one PR though

Reply:
I think in external provisioner workflow, we cannot capture driver errors regardless because the error event is generated by the external-provisioner, not the in-tree controller.

For now let's not touch the existing error metrics and just focus on adding a new total latency metric.

I'm still not convinced we should remove OperationCompletionHook yet. They measure two different things, we may find them useful for different purposes. If we do decide to remove it, we would need to follow the deprecation policy which is one year.

Reply:
done